packages/odyssey-react-mui/src/labs/OdysseyPickers/PickerWithOptionAdornment.tsx

```diff
@@ -0,0 +1,430 @@
+  OptionMetadataComponent,
+  OptionProps,
+} from "./BasicPicker";
+import { Heading6 } from "../../Typography";
+import { Tag } from "../../Tag";
+
+type Adornment = ReactNode | string;
+
+type AdornmentLabelDescription = LabelDescription & {
+  adornment: Adornment;
+};
+
+type AdornmentLabelDescriptionMetadata = AdornmentLabelDescription & Metadata;
+
+export type AdornmentOptionType =
+  | AdornmentLabelDescription
+  | AdornmentLabelDescriptionMetadata;
+
+const OptionAdornmentContainer = styled("div", {
+  shouldForwardProp: (prop) =>
+    prop !== "odysseyDesignTokens" &&
+    prop !== "adornmentSize" &&
+    prop !== "isTagContainer",
+})<{
+  adornmentSize?: AdornmentSize;
+  isTagContainer?: boolean;
+  odysseyDesignTokens: DesignTokens;
+}>(
+  ({
+    adornmentSize = "small",
+    isTagContainer = false,
+    odysseyDesignTokens,
+  }) => ({
+    position: "relative",
+    // push icon up by one px for better visual alignment
+    bottom: "1px",
+    alignSelf: "flex-start",
+    width: odysseyDesignTokens.Spacing5,
+    height: odysseyDesignTokens.Spacing5,
+    overflow: "hidden",
+    marginInlineEnd: odysseyDesignTokens.Spacing3,
+
+    svg: {
+      width: "100%",
+      height: "auto",
+    },
+
+    img: {
+      position: "absolute",
+      top: "50%",
+      left: "50%",
+      width: "100%",
+      transform: "translate(-50%, -50%)",
+    },
+
+    ...(adornmentSize === "large" &&
+      !isTagContainer && {
+        bottom: 0,
+        width: odysseyDesignTokens.Spacing8,
+        height: odysseyDesignTokens.Spacing8,
+      }),
+
+    ...(isTagContainer && {
+      bottom: 0,
+      alignSelf: "center",
+      width: odysseyDesignTokens.Spacing4,
+      height: "auto",
+      maxHeight: odysseyDesignTokens.Spacing4,
+      marginInlineEnd: odysseyDesignTokens.Spacing2,
+
+      svg: {
+        display: "flex",
+        width: "100%",
+        height: "auto",
+      },
+    }),
+  }),
+);
+
+type OptionAdornmentProps = {
+  adornment: Adornment;
+  adornmentSize: AdornmentSize;
+  odysseyDesignTokens: DesignTokens;
+};
+
+const OptionAdornment = ({
+  adornment,
+  adornmentSize,
+  odysseyDesignTokens,
+}: OptionAdornmentProps) => {
+  const isImageAdornment = typeof adornment === "string";
+
+  if (isImageAdornment) {
+    return (
+      <OptionAdornmentContainer
+        adornmentSize={adornmentSize}
+        odysseyDesignTokens={odysseyDesignTokens}
+      >
+        {/* NOTE: Intentionally leaving alt as an empty string here so screen readers will ignore this image */}
+        {/* Image should be suffciently described by the adjacent title and/or description of the option */}
+        <img src={adornment} alt="" role="presentation" />
+      </OptionAdornmentContainer>
+    );
+  } else {
+    return (
+      <OptionAdornmentContainer
+        adornmentSize={adornmentSize}
+        odysseyDesignTokens={odysseyDesignTokens}
+      >
+        <>{adornment}</>
+      </OptionAdornmentContainer>
```

Comment:
We could just do this right?
```suggestion
      <OptionAdornmentContainer
        adornmentSize={adornmentSize}
        odysseyDesignTokens={odysseyDesignTokens}
      >
        {adornment}
      </OptionAdornmentContainer>
```

If not, we should change to `ReactNode` instead of `ReactElement`.